include/cop0.h

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file cop0.h
+ * @brief N64 COP0 Interface
+ * @ingroup n64sys
+ */
+
+/**
+ * @addtogroup n64sys
+ * @{
+ */
+
+#ifndef __LIBDRAGON_COP0_H
+#define __LIBDRAGON_COP0_H
+
+/** @brief Read the COP0 Count register (see also TICKS_READ). */
+#define C0_COUNT() ({ \
+    uint32_t x; \
+    asm volatile("mfc0 %0,$9":"=r"(x)); \
+    x; \
+})
+
+/** @brief Write the COP0 Count register. */
+#define C0_WRITE_COUNT(x) ({ \
+    asm volatile("mtc0 %0,$9"::"r"(x)); \
+})
+
+
+/** @brief Read the COP0 Compare register. */
+#define C0_COMPARE() ({ \
+    uint32_t x; \
+    asm volatile("mfc0 %0,$11":"=r"(x)); \
+    x; \
+})
+
+/** @brief Write the COP0 Compare register. */
+#define C0_WRITE_COMPARE(x) ({ \
```

Comment:
I think we should wrap these with `do { ... } while (0)`, What do you think?

Reply:
The statement expression `({ ... })` is a GNU extension that has the same effect and also allows to return a value. I'd settle for using it everywhere in macros because it's more powerful.